documentation/config_file_auth_migration.md

```diff
@@ -0,0 +1,267 @@
+              * A single string, representing the request's "name" derived from the Common Name (CN) attribute within an X.509 certificate's Subject Distinguished Name (DN). This string can be an exact name, a glob, or a regular expression.
+              * A single map value containing an `extension` key.
+              * A single map value containing a `certname` key.
+              * An array of values, including string and map values.
+          * An `allow-unauthenticated` value, but if present, there cannot also be an `allow` value.
+
+For an full example, see the [HOCON `auth.conf` documentation](./config_file_auth.html#hocon-example).
+
+### Converting a simple rule
+
+Let's convert this simple deprecated `auth.conf` authorization rule:
+
+```
+path /puppet/v3/environments
+method find
+allow *
+```
+
+We'll start with a skeletal HOCON `auth.conf` file:
+
+``` hocon
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path:
+                type:
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ]
+}
+```
+
+Next, let's convert each component of the deprecated rule in the new HOCON format.
+
+1. Add the path to the new rule's [`path`](./config_file_auth.html#match-request) setting in its `match-request` section.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type:
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+2. Next, add its type to the section's [`type`](./config_file_auth.html#match-request) setting. Since this is a literal string path, the type is `path`.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+3. The legacy rule has a [`method`](./config_file_auth.html#method-1) setting, with an indirector value of `find` that's equivalent to the GET and POST HTTP methods. We can implement these by adding an optional HOCON [`method`](./config_file_auth.html#match-request) setting in the rule's `match-request` section and specifying GET and POST as an array.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+4. Next, set the [`allow`](#allow-allow-unauthenticated-and-deny) setting. The legacy rule used a `*` glob, which is also supported in HOCON.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
```

Comment:
The asterisk needs to be wrapped inside of double quotes, so like:

```
allow: *
```

Asterisk is a reserved character in HOCON so you can't just have the value be * directly.